compiler/rustc_errors/src/lib.rs

```diff
@@ -187,29 +187,40 @@ impl SubstitutionPart {
 impl CodeSuggestion {
     /// Returns the assembled code suggestions, whether they should be shown with an underline
     /// and whether the substitution only differs in capitalization.
-    pub fn splice_lines(&self, sm: &SourceMap) -> Vec<(String, Vec<SubstitutionPart>, bool)> {
+    pub fn splice_lines(
+        &self,
+        sm: &SourceMap,
+    ) -> Vec<(String, Vec<SubstitutionPart>, Vec<Vec<(usize, usize)>>, bool)> {
         use rustc_span::{CharPos, Pos};
 
         fn push_trailing(
             buf: &mut String,
             line_opt: Option<&Cow<'_, str>>,
             lo: &Loc,
             hi_opt: Option<&Loc>,
-        ) {
+        ) -> usize {
+            let mut count = 0;
```

Comment:
```suggestion
            let mut line_count = 0;
```
Looks like we are counting lines here and below, but it's far from being clear.

Reply:
We are counting the lines that the inserted snippet is adding to the output. Adding a comment and renaming.